lib/task-data/schemas/diag-update-spirom.json

```diff
@@ -0,0 +1,33 @@
+{
+    "copyright": "Copyright 2017, Dell EMC, Inc.",
+    "definitions": {
+        "imagePath": {
+            "description": "Image zip file path",
+            "type": "string",
+            "pattern": "^/.*\\.zip$"
+        },
+        "imageMode": {
+            "description": "Update mode",
+            "type": "string",
+            "enum": ["fullbios", "bios", "uefi", "serdes", "post", "me", "0", "1", "2", "3",
```

Comment:
Since the integer format `imageMode` is very hard to remember and distinguish, if this is not a must-have request, I'd suggest you removing the supporting on this. Using the named `imageMode` I think it's enough.